ckan/config/middleware/__init__.py

```diff
@@ -151,6 +151,10 @@ def __call__(self, environ, start_response):
 
         log.debug('Serving request via {0} app'.format(app_name))
         environ['ckan.app'] = app_name
+        # no SCRIPT_NAME provided by nginx unit for old wsgi
+        if 'SCRIPT_NAME' not in environ:
```

Comment:
I have no idea how not have SCRIPT_NAME can influence the rest of the application. This needs to be tested well before going forward.

Reply:
According to wsgi standard, `SCRIPT_NAME` is mandatory, even though is the empty string. nginx-unit apparently doesn't think this way, so we'll have to set defaults in ckan. ckan assumes there will be a `SCRIPT_NAME` set.